tests/integration/components/new-signup/input-test.js

```diff
@@ -136,4 +136,62 @@ module('Integration | Component | new-signup/input', function (hooks) {
 
     assert.dom('[data-test-button="signup"]').isDisabled();
   });
+
+  test('input field should remove spaces if user pastes text', async function (assert) {
+    assert.expect(1);
+
+    this.setProperties({
+      currentStep: 'firstName',
+      onChange: (step, value) => {
+        this.inputValue = value;
+      },
+      onClick: () => {},
+    });
+
+    await render(hbs`
+      <NewSignup::Input
+        @currentStep={{this.currentStep}}
+        @onChange={{this.onChange}}
+        @onClick={{this.onClick}}
+      />
+    `);
+
+    const input = this.element.querySelector('[data-test-signup-form-input]');
+    input.value = 'John Doe';
+    await triggerEvent(input, 'input');
+
+    assert.strictEqual(
+      this.inputValue,
+      'JohnDoe',
+      'Spaces should be removed from input',
+    );
```

Comment:
Can we add another assertion here to check the value in actual input?
assert.dom(input).hasValue('JohnDoe')